python/ray/autoscaler/v2/instance_manager/subscribers/threaded_ray_installer.py

```diff
@@ -31,65 +46,50 @@ def __init__(
         self._max_concurrent_installs = max_concurrent_installs
         self._max_install_attempts = max_install_attempts
         self._install_retry_interval = install_retry_interval
+        self._error_queue = error_queue
         self._ray_installation_executor = ThreadPoolExecutor(
             max_workers=self._max_concurrent_installs
         )
 
     def notify(self, events: List[InstanceUpdateEvent]) -> None:
         for event in events:
-            if event.new_instance_status == Instance.ALLOCATED:
+            if event.new_instance_status == Instance.RAY_INSTALLING:
                 self._install_ray_on_new_nodes(event.instance_id)
 
     def _install_ray_on_new_nodes(self, instance_id: str) -> None:
         allocated_instance, _ = self._instance_storage.get_instances(
             instance_ids={instance_id},
-            status_filter={Instance.ALLOCATED},
+            status_filter={Instance.RAY_INSTALLING},
         )
         for instance in allocated_instance.values():
+            assert instance.node_kind == NodeKind.WORKER
             self._ray_installation_executor.submit(
                 self._install_ray_on_single_node, instance
             )
 
     def _install_ray_on_single_node(self, instance: Instance) -> None:
-        assert instance.status == Instance.ALLOCATED
-        success, version = self._instance_storage.upsert_instance(
-            instance, expected_instance_version=instance.version
```

Comment:
`ThreadedRayInstaller` should not do `instance_storage.upsert_instance`. That's the responsibility of the autoscaler v2 reconciler; otherwise, we will have version mismatch errors.